### ISSUE TITLE
Allow fully random ports for Envoy

### ISSUE THREAD

**Issue by codefromthecrypt** [istio/istio#32184]
**Describe the feature request**
In getenvoy, we have unit and end-to-end tests that would ideally run in parallel. One test is envoy bootstrap, and I noticed that while the envoy admin port can be specified, and pilot's addresses can be set to "127.0.0.1:0" others in the envoy template are hard-coded, notably ports 15020 15021 and 15090. While this isn't a production issue, this also limits the amount of testing that someone can do to serial or indirect ports via running inside docker.

I'd love to see the following changes to tools/packaging/common/envoy_bootstrap.json in descending priority:
* all ports that refer to core services templatable
* some solution that allows port 0 at least for the envoy admin port (possibly reading back with --admin-address-path when on the same host eg mock mode)

**Describe alternatives you've considered**
running serially, running tests, mapping pilot and envoy ports with a docker container driver. These are slow and can lead to flakes especially the former can be flakey as users have to stop all related services on their hosts to avoid effecting behavior.

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[x] Developer Infrastructure

**Additional context**

**Comment by howardjohn** [comment]
This is a reasonable request but its a duplicate of https://github.com/istio/istio/issues/31138 so lets track there. Thanks!
